internal/datastore/proxy/hedging.go

```diff
@@ -276,24 +292,20 @@ type hedgingTupleQuery struct {
 type tupleExecutor func(ctx context.Context) (datastore.TupleIterator, error)
 
 func executeQuery(ctx context.Context, exec tupleExecutor, queryHedger hedger) (delegateIterator datastore.TupleIterator, err error) {
-	doneLock := sync.Mutex{}
-	alreadyReturned := false
-
+	var once sync.Once
 	subreq := func(ctx context.Context, responseReady chan<- struct{}) {
 		tempIterator, tempErr := exec(ctx)
-		doneLock.Lock()
-		defer doneLock.Unlock()
-
-		if alreadyReturned {
-			if tempErr == nil {
-				tempIterator.Close()
-			}
-		} else {
-			alreadyReturned = true
+		executed := false
```

Comment:
Maybe rename `executed` to `resultsUsed` or something? As it is it's a little tricky to reason about why they don't all end up with `executed = true`.

Reply:
used the name you suggested + added a comment about it